src/utils/Types.ts

```diff
@@ -4,31 +4,39 @@ import type { Difficulty } from './Difficulty';
 
 export type RiddlePreview = {
 	id?: number;
+	//TODO: Make displays based on linkId instead of dbId
+	linkId: string;
 	name: string;
 	image?: string;
 	state: RiddleStatus;
-	countryCode: CountryCode;
+	language: CountryCode;
 	difficulty: Difficulty;
 };
 
-export type RiddleUpsertDetail = RiddlePreview & {
+export type RiddleUpsertDetail = Omit<RiddlePreview, 'state'> & {
 	description: string;
 	solvedText: string;
 	solvedImage?: string;
 	questions: QuestionUpsertDetail[];
+	questionOrder?: 'sequence' | 'parallel';
+	sharingInformation: SharingInformationUpsert;
 };
 
 export type RiddleDisplayDetail = Omit<RiddleUpsertDetail, 'questions'> & {
 	questions: QuestionDisplayDetail[];
 };
 
+type TextType = {
+	text: string;
```

Comment:
This was necessary because form library can work only with objects, not plain strings. We could alternatively use just unnamed `{text: string}`.